src/designspace.rs

```diff
@@ -0,0 +1,213 @@
+//! Reading and writing designspace files.
+
+#![deny(rustdoc::broken_intra_doc_links)]
+
+use quick_xml::de::from_reader;
+use std::{fs::File, io::BufReader, path::Path};
+
+use crate::error::DesignSpaceLoadError;
+
+/// A [designspace]].
+///
+/// [designspace]: https://fonttools.readthedocs.io/en/latest/designspaceLib/index.html
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "designspace")]
+pub struct DesignSpaceDocument {
+    /// Design space format version.
+    pub format: f32,
+    /// One or more axes.
+    pub axes: Axes,
+    /// One or more sources.
+    pub sources: Sources,
+    /// One or more instances.
+    pub instances: Instances,
+}
+
+/// https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#axes-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "axes")]
+pub struct Axes {
+    /// One or more axis definitions.
+    pub axis: Vec<Axis>,
+}
```

Comment:
Yea, I think we want some kind of custom derive here, to just get a `Vec<Axis>`.

Reply:
I wasn't sure how but it seems https://stackoverflow.com/questions/58571896/rust-serde-deserialize-xml-directly-to-vect offers a way